pkg/measurement/measurement.go

```diff
@@ -32,90 +32,74 @@ type measurement struct {
 
 	p *properties.Properties
 
-	opMeasurement map[string]ycsb.Measurement
+	measurer ycsb.Measurer
 }
 
-func (m *measurement) measure(op string, lan time.Duration) {
-	m.RLock()
-	opM, ok := m.opMeasurement[op]
-	m.RUnlock()
-
-	if !ok {
-		opM = newHistogram(m.p)
-		m.Lock()
-		m.opMeasurement[op] = opM
-		m.Unlock()
-	}
-
-	opM.Measure(lan)
+func (m *measurement) measure(op string, start time.Time, lan time.Duration) {
+	m.Lock()
+	m.measurer.Measure(op, start, lan)
+	m.Unlock()
 }
 
 func (m *measurement) output() {
 	m.RLock()
 	defer m.RUnlock()
-	keys := make([]string, len(m.opMeasurement))
-	var i = 0
-	for k := range m.opMeasurement {
-		keys[i] = k
-		i += 1
-	}
-	sort.Strings(keys)
 
-	lines := [][]string{}
-	for _, op := range keys {
-		line := []string{op}
-		line = append(line, m.opMeasurement[op].Summary()...)
-		lines = append(lines, line)
+	outFile := m.p.GetString(prop.MeasurementRawOutputFile, "")
+	var w *bufio.Writer
+	if outFile == "" {
+		w = bufio.NewWriter(os.Stdout)
+	} else {
+		f, err := os.Create(outFile)
+		if err != nil {
+			panic("failed to create output file: " + err.Error())
+		}
```

Comment:
We'd better add `defer f.Close()` here, although it's only called once via ycsb command.

Reply:
Added, thanks!